### DISCUSSION TITLE
Кнопка QueryTag

### DISCUSSION THREAD

**Discussion by k-rustemuly** [moonshine-software/moonshine#662] [Ideas]
Если QueryTag кнопка активна, то при повторном нажатии удалить параметры query-tag из href. Тогда повторная нажатия кнопки просто вернет все записи

**Comment by k-rustemuly** [comment]
Я в самом MoonShine\Buttons\QueryTagButton.php
```php
->when(
    $tag->isActive(),
    fn (ActionButton $btn): ActionButton => $btn
        ->primary()
        ->customAttributes([
            'class' => 'active-query-tag',
            'href' => to_page(page: IndexPage::class, resource: $resource),
        ])

)
```
протестировал так, удобно

**Reply by lee-to** [reply]
Можешь сделать pr, только не используй IndexPage а бери по типу $resource->getPages()->findByType

**Reply by k-rustemuly** [reply]
ок

**Comment by DissNik** [comment]
Только есть default() и кнопка будет всегда активна

**Reply by k-rustemuly** [reply]
Мне надо при повторном нажатии кнопки выводились все записи

**Reply by DissNik** [reply]
я понял это, просто default() позволяет сделать кнопку все записи, или какой-то QueryTag активным по умолчанию, даже если параметр отсутствует, поэтому такую кнопку отключить нельзя будет, но одно другому не мешает

**Reply by lee-to** [reply]
Вы обсуждаете уже слитый pr?)

**Reply by k-rustemuly** [reply]
я же сразу создал pr. После [коммента](https://github.com/orgs/moonshine-software/discussions/662#discussioncomment-7692573) )
